pages/docs/recipes/adding-a-left-panel.mdx

```diff
@@ -0,0 +1,320 @@
+- We're attaching the extension named `bed-management-left-panel` to the slot named `bed-management-left-panel-slot` when the `LeftPanel` component mounts. Similarly, we're detaching the extension from the same extension slot when the component unmounts. This allows other components to add links to the left panel by adding their own content to this slot.
+- We're using the `<ExtensionSlot>` component to render the content of the `bed-management-left-panel-slot` slot dynamically into the `LeftPanel` component.
+
+### Step 2: Connect the `LeftPanel` component to your `Root` component
+
+To do so, you'll want to add your `LeftPanel` component to your `BrowserRoutes` definition to ensure it gets rendered on all the routes you've configured in your app. Here's the example `Root` component after adding the `LeftPanel`:
+
+```tsx copy {14-20, 24}
+// src/root.component.tsx
+import React, { useEffect } from "react";
+import { BrowserRouter, Route, Routes } from "react-router-dom";
+import { setLeftNav, unsetLeftNav } from "@openmrs/esm-framework";
+import BedAdministrationTable from "./bed-administration/bed-administration-table.component";
+import Home from "./home.component";
+import LeftPanel from "./left-panel/left-panel.component";
+import WardWithBeds from "./ward-with-beds/ward-with-beds.component";
+import styles from "./root.scss";
+
+const Root: React.FC = () => {
+  const spaBasePath = window.spaBase;
+
+  useEffect(() => {
+    setLeftNav({
+      name: "bed-management-left-panel-slot",
+      basePath: spaBasePath,
+    });
+    return () => unsetLeftNav("bed-management-left-panel-slot");
+  }, [spaBasePath]);
+
+  return (
+    <BrowserRouter basename={`${window.getOpenmrsSpaBase()}bed-management`}>
+      <LeftPanel />
+      <main className={styles.container}>
+        <Routes>
+          <Route path="/summary" element={<Home />} />
+          <Route path="/location/:location" element={<WardWithBeds />} />
+          <Route path="/administration" element={<BedAdministrationTable />} />
+        </Routes>
+      </main>
+    </BrowserRouter>
+  );
+};
+
+export default Root;
+```
+
+Key things to note here are:
+
+- We're registering our slot with the `LeftNav` store when the component mounts, setting the provided spaBasePath (`/openmrs/spa`) as the basePath. We're also unregistering the slot from the LeftNav store when the component unmounts. The [LeftNav store](https://github.com/openmrs/openmrs-esm-core/blob/da401641b4b1895e593e9bcea73af8f8dbd505e5/packages/framework/esm-styleguide/src/left-nav/index.tsx#L18) is essentially a Zustand store that tracks slot names and their associated `basePath`s.
+- We're adding the `LeftPanel` component at the top of the `<BrowserRouter>` tree, just outside the routes declarations. This means that the `LeftPanel` gets rendered in all the pages defined in this app.
```

Comment:
I'm being a bit pedantic on this point, but I think we need to consistently use "app" (collection of related extensions, pages, and functionality), "page" (a MF that is registered as a page), etc.

```suggestion
- We're adding the `LeftPanel` component at the top of the `<BrowserRouter>` tree, just outside the routes declarations. This means that the `LeftPanel` gets rendered in all the screens that use this page.
```